modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -185,6 +185,8 @@ enum VideoCaptureProperties {
        CAP_PROP_ORIENTATION_AUTO=49, //!< if true - rotates output frames of CvCapture considering video file's metadata  (applicable for FFmpeg back-end only) (https://github.com/opencv/opencv/issues/15499)
        CAP_PROP_HW_ACCELERATION=50, //!< (**open-only**) Hardware acceleration type (see #VideoAccelerationType). Setting supported only via `params` parameter in cv::VideoCapture constructor / .open() method. Default value is backend-specific.
        CAP_PROP_HW_DEVICE      =51, //!< (**open-only**) Hardware device index (select GPU if multiple available)
+       CAP_PROP_AUDIO_ENABLE =1000,
```

Comment:
> // Select audio or video

We should support processing of both video and audio streams. This is why we are integrating into `VideoCapture` instead of adding dedicated Audio API.

The next requirement is that video and audio must be synchronized in a some way (need to define these details and write them somewhere. Other "backends" and user apps should follow there guidelines).

Also we need these properties:

- (Priority 0) CAP_PROP_VIDEO_STREAM - (**open-only**, 0-based index or -1 to disable) - Default value is 0.
- (Priority 0) CAP_PROP_AUDIO_STREAM - (**open-only**, 0-based index or -1 to disable) - Specify stream in  multi-language media files. -1 - disable audio processing (default)
- (Priority 1) CAP_PROP_AUDIO_POS - (**read-only**, in samples) accurate audio sample timestamp of previous grabbed fragment. See CAP_PROP_AUDIO_SAMPLES_PER_SECOND.
- (Priority 1) CAP_PROP_AUDIO_DATA_DEPTH - (**open-only**, Mat::depth()) Default value is -1: use "native" file/codec information. Alternative definition to bits-per-sample, but with clear handling of 32F / 32S.
- (Priority 1) CAP_PROP_AUDIO_SAMPLES_PER_SECOND - (**open-only**) 0 - determine from file/codec input. If not specified through parameters or input, then selected audio sample rate is 44100. Note: resampling is not performed by OpenCV.
- (Priority 2) CAP_PROP_AUDIO_CHANNELS - (**open-only**?, bitset) - to properly handle stereo or 5.1 streams. ? 0 - use average of all channels. Use -1 to grab all channels. Default value is 0 / -1 ?
- (Priority 3) CAP_PROP_AUDIO_TOTAL_SAMPLES - (**read-only**) Number of samples in the file. Returns zero if information is not available (live streams).
- (Priority 3) CAP_PROP_AUDIO_TOTAL_CHANNELS - (**read-only**) Number of audio channels in the selected audio stream.
- (Priority 4) CAP_PROP_AUDIO_TOTAL_STREAMS - (**read-only**) Number of audio stream in the used media.

* 44100 has problem with 24 FPS video (1827.5 audio samples per frame). 48000 is good for 24,25,30 FPS media streams, but it has less support from codecs. See https://en.wikipedia.org/wiki/Sampling_(signal_processing)

**TBD later**